MiniAn/minian_run.py

```diff
@@ -11,555 +11,32 @@
 
 sys.path.append('..')
 import utilities as utils
-import minian_utilities as mn_utils
-
-# Import for MiniAn lib
-from minian.utilities import TaskAnnotation, get_optimal_chk, custom_arr_optimize, save_minian, open_minian
-from minian.preprocessing import denoise, remove_background
-from minian.initialization import seeds_init, pnr_refine, ks_refine, seeds_merge, initA, initC
-from minian.cnmf import compute_trace, get_noise_fft, update_spatial, update_temporal, unit_merge, update_background, compute_AtC
-from minian.motion_correction import apply_transform, estimate_motion
+import minian_main        as mn_main
+import minian_parameters  as mn_params
+import minian_definitions as mn_defs
 
 # Import for PyInstaller
 from multiprocessing import freeze_support
 freeze_support()
 
-# Text definitions
-START_CLUSTER       = "Starting cluster..."
-ADVANCED_BAD_TYPE   = "One of the advanced settings is not of a python type"
-LOAD_DATA           = "Loading dataset to MiniAn..."
-ONE_PARM_WRONG_TYPE = "One parameter of {0} function is of the wrong type"
-NO_CELLS_FOUND      = "No cells where found"
-PREPROCESS          = "Pre-processing..."
-PREPROC_REMOVE_GLOW = "Pre-processing: removing glow..."
-PREPROC_DENOISING   = "Pre-processing: denoising..."
-PREPROC_REMOV_BACKG = "Pre-processing: removing background..."
-PREPROC_SAVE        = "Pre-processing: saving..."
-CORRECT_MOTION_ESTIM_SHIFT  = "Correcting motion: estimating shifts..."
-CORRECT_MOTION_APPLY_SHIFT  = "Correcting motion: applying shifts..."
-PREP_DATA_INIT              = "Preparing data for initialization..."
-INIT_SEEDS                  = "Initializing seeds..."
-INIT_SEEDS_PNR_REFI         = "Initializing seeds: PNR refinement..."
-INIT_SEEDS_KOLSM_REF        = "Initializing seeds: Kolmogorov-Smirnov refinement..."
-INIT_SEEDS_MERG             = "Initializing seeds: merging..."
-INIT_COMP                   = "Initializing components..."
-INIT_COMP_SPATIAL           = "Initializing components: spatial..."
-INIT_COMP_TEMP              = "Initializing components: temporal..."
-INIT_COMP_MERG              = "Initializing components: merging..."
-INIT_COMP_BACKG             = "Initializing components: background..."
-CNMF_IT                     = "CNMF {0} iteration"
-CNMF_ESTIM_NOISE            = CNMF_IT + ": estimating noise..."
-CNMF_UPDAT_SPATIAL          = CNMF_IT + ": updating spatial components..."
-CNMF_UPDAT_BACKG            = CNMF_IT + ": updating background components..."
-CNMF_UPDAT_TEMP             = CNMF_IT + ": updating temporal components..."
-CNMF_MERG_COMP              = CNMF_IT + ": merging components..."
-CNMF_SAVE_INTERMED          = CNMF_IT + ": saving intermediate results..."
-SAVING_FINAL                = "Saving final results..."
-SAVING_TO_DORIC             = "Saving data to doric file..."
-
-kwargs = {}
-params_doric = {}
 danse_parameters = {}
 
 try:
     for arg in sys.argv[1:]:
         exec(arg)
```

Comment:
Can we have something like `danse_parameters = eval(arg)`? (Does it work like that?) I feel like it's not clear from the code where the parameters come from.

Reply:
I have to check that.

Reply:
or `exec('danse_parameters = ' + arg)`

Reply:
Perfect!